chirp/library/do_delete_audio_file_from_db.py

```
@@ -0,0 +1,145 @@
+        except:
+            raise
+        else:
+            self.del_rows(fingerprints, table="audio_files")
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='Delete records in audio file sqlite database.')
+    parser.add_argument(
+        'fingerprint', type=str, nargs='+',
+        help=(
+            'Audio file finterprint to delete. Can specify more '
+            'than one, space delimited.  Must pass the --delete flag to '
+            'confirm the delete.'))
+    parser.add_argument(
+        '--delete', action="store_true",
+        help='Delete the audiofile based on the fingerprint passed in')
+    parser.add_argument(
+        '--db', action='store', type=str, default=None,
+        help=(
+            "Specify a full filesystem path to the database file "
+            "to operate on."))
+
+    args = parser.parse_args()
+
+    afm = AudioFileManager(library_db_file=args.db)
+
+    sys.stdout.write("ARGS: {} \n\n".format(str(args)))
+    sys.stdout.write("Using database: {}\n".format(afm.db_path))
+
+    fingerprints = args.fingerprint
+    try:
+        tags = afm.get_tags(fingerprints)
+
+        audio_files = list(afm.get_audio_files(fingerprints))
+        sys.stdout.write("\nROWS TO DELETE from audio_files\n\n")
+        afm.print_rows(audio_files)
+
+        sys.stdout.write("\nROWS TO DELETE from id3_tags\n\n")
+        afm.print_rows(tags)
+
+        if set(fingerprints) != set(f["fingerprint"] for f in audio_files):
+            sys.stdout.write(
+                "\n\nWARNING: A fingerprint was given that does not match "
+                "an audio file in the database\n\n")
+
+        if args.delete:
+            afm.del_audiofiles(fingerprints)
+            sys.stdout.write("\nDELETED\n")
+        else:
+            sys.stdout.write("\nNOTHING DELETED.  Pass in the --delete flag.\n")
+    except Exception as e:
+        raise
+        sys.stderr.write("ERROR: {}\n".format(str(e)))
```

I'm not sure if you meant to leave this in here but it's never executed (right?). Anyway, I don't think you should try to catch an exception; just let it raise. When someone encounters an error, I'd rather get a bug report with a complete traceback.

---

Thanks for catching that.  I meant to reverse the order of those of the raise and the stderr.write statements.  I've removed the entry try/except block so the code will just raise an exception as normal.